packages/theme/src/unified/types.ts

```diff
@@ -14,6 +14,12 @@
  * limitations under the License.
  */
 
+import { Theme as Mui4Theme } from '@material-ui/core/styles';
+import { Theme as Mui5Theme } from '@mui/material/styles';
+
+export type SupportedVersions = 'v4' | 'v5';
+export type SupportedThemes = Mui4Theme | Mui5Theme;
```

Comment:
a bit "stronger" types. After fixing the existing issues it might be worth changing this ot an `enum` - `Record<>` type to automatically convert the return depending on the version requested.